networking/src/Network/Transport/Abstract.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE DeriveDataTypeable         #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes                 #-}
+
+-- | Network Transport
+module Network.Transport.Abstract
+  ( -- * Types
+    Transport(..)
+  , EndPoint(..)
+  , Connection(..)
+  , Event(..)
+  , QDisc(..)
+  , NT.ConnectionId
+  , NT.ConnectionBundle
+  , NT.Reliability(..)
+  , NT.EndPointAddress(..)
+    -- * Hints
+  , NT.ConnectHints(..)
+  , NT.defaultConnectHints
+    -- * Error codes
+  , NT.TransportError(..)
+  , NT.NewEndPointErrorCode(..)
+  , NT.ConnectErrorCode(..)
+  , NT.SendErrorCode(..)
+  , NT.EventErrorCode(..)
+  , EventError(..)
+  , hoistTransport
+  , hoistEndPoint
+  , hoistConnection
+  ) where
```

Comment:
4 space indentation: consider setting up `editorconfig` to get it to pick up project indentation automatically :)